src/KeyringController.ts

```diff
@@ -898,10 +897,12 @@ class KeyringController extends EventEmitter {
 
         // This call is required on the first call because encryptionKey
         // is not yet inside the memStore
-        this.memStore.updateState({
-          encryptionKey,
-          encryptionSalt,
-        });
+        this.memStore.putState(
+          Object.assign(this.memStore.getState(), {
+            encryptionKey,
+            encryptionSalt,
+          }),
+        );
```

Comment:
Nit: Is this change necessary? This seems equivalent to what was here before

```suggestion
        this.memStore.updateState({
          encryptionKey,
          encryptionSalt,
        });
```

Reply:
I think it would raise a type error because encryptionKey and encryptionSalt could be undefined, which doesn't mean that the two keys will be removed from the state

Reply:
With `putState` and an object assign we pass the current state and encryptionKey and encryptionSalt _if they are defined_

Reply:
It should already be impossible for them to be `undefined` here. This block is when `cacheEncryptionKey` is set. A few lines above here we already verify that `encryptionKey` is a string and `encryptionSalt` matches the salt in the encrypted vault

Reply:
I rolled back, adding a non-null assertion on the `encryptionSalt`